### PR TITLE
Use standalone ArrowTypes package

### PR DESCRIPTION
**Pull request by omus**
Fixes: https://github.com/JuliaData/Arrow.jl/issues/209.

The one big hangup with this change is the inclusion of the `Manifest.toml`. The manifest allows use to dev `src/ArrowTypes` which means any changes to that code will immediately be reflected during local development. Additionally, since the manifest is committed the CI should also work the same way. Unfortunately, this can lead to changes to ArrowTypes.jl going unregistered (like what has currently happened).

Alternatively, we may want to not commit the Manifest.toml file and instead document how to setup a manifest file for local development and never commit it. This approach is more setup for developers but ensures that the CI is testing what users of the packages will ultimately end up seeing. I'm in favor of this approach

### CODE REVIEW

**Comment by omus** [comment]
As 1.3 seems to be failing on the Manifest.toml I'll switch this over to using the other idea

**Comment by quinnj** [comment]
Yes, I'm in favor of not committing the manifest file

**Comment by omus** [comment]
Expecting CI to fail until #213 is merged and registered

File: .github/workflows/ci.yml
Comment (omus) [inline .github/workflows/ci.yml:62]: Including the package name as part of the matrix would eliminate the YAML redundancy but this is the one part where doing that is hard. It may be possible to use symlinks to clean this up

Comment (omus) [reply .github/workflows/ci.yml]: In fbbd33c19fcfa522de3cd6b58fbd33694935d85a I've switched over to using symlinks. I'm specifying `working-directory` on each step currently which has the best chance of working but I'll attempt to use `defaults` once I validate GHA works with the symlinks

File: Project.toml
Comment (omus) [inline Project.toml:21]: The Arrow.jl package should only requires ArrowTypes.jl 1.0 but the tests require 1.1.

**Comment by omus** [comment]
@quinnj please approve the CI to run when you get the chance. There may be a little debugging to do on the CI side before this is ready

**Comment by quinnj** [comment]
I don't have an option to approve CI; that would seem to indicate something in the ci.yml file that's causing CI not to run

**Comment by ericphanson** [comment]
> something in the ci.yml file that's causing CI not to run 

If I edit it in the GitHub editor, it has syntax highlighting and shows errors:

Screenshot 2021-06-25 at 19 44 23

File: .github/workflows/ci.yml
Comment (omus) [reply .github/workflows/ci.yml]: The `working-directory` solution [was a no-go](https://github.com/JuliaData/Arrow.jl/pull/212#issuecomment-868731166). Using `JULIA_PROJECT` now.

**Comment by omus** [comment]
RTM. CI is now functional

File: codecov.yaml
Comment (omus) [inline codecov.yaml:5]:
```suggestion
    after_n_builds: 6

```

File: README.md
Comment (ericphanson) [inline README.md:32]:
```suggestion
When developing on Arrow.jl it is recommended that you run the following to ensure that any
```

**Review by ericphanson — APPROVED** [review]

**Review by jrevels — APPROVED** [review]
LGTM. What are the follow ups for registration?

**Comment by omus** [comment]
> LGTM. What are the follow ups for registration?

Should just be the registration. ArrowTypes version 1.1.0 has already [been registered](https://github.com/JuliaRegistries/General/pull/39003). Once this change is merged and registered any package which only depends on ArrowTypes.jl should be able to extend Arrow.jl without having to use Arrow.jl as a dependency.

**Comment by omus** [comment]
I'm just validating that adding Arrow support through ArrowTypes.jl is fully working

**Comment by omus** [comment]
> I'm just validating that adding Arrow support through ArrowTypes.jl is fully working

We're good here. My validation brought up some things I need to work through which are independent from these changes.

**Pull request merged**
